Mathlib/CategoryTheory/Enriched/FunctorCategory.lean

```diff
@@ -335,4 +346,110 @@ noncomputable def isLimitConeFunctorEnrichedHom :
 
 end
 
+/-- The identity for the `J ⥤ V`-enrichment of the category `J ⥤ C`. -/
+@[simps]
+noncomputable def functorEnrichedId [HasFunctorEnrichedHom V F₁ F₁] :
+    𝟙_ (J ⥤ V) ⟶ functorEnrichedHom V F₁ F₁ where
+  app j := enrichedId V _
+  naturality j j' f := by
+    dsimp
+    ext k
+    dsimp
+    rw [assoc, assoc, id_comp, enrichedId_π]
+    erw [end_.lift_π]
```

Comment:
I'm happy with the answer "No, this is an unreasonable ask", but:

Could I request that whenever we introduce an `erw`, we *also* include a comment explaining either why it is unavoidable, or better than alternatives?

Optimistically, this will result in some `erw` not happening, as writing the note might inspire a solution.

And otherwise, it means that we won't have to come back to the same `erw` again and again and work out the explanation before forgetting it again.

Reply:
It seems at this `erw` that the problem is that before the `erw` the composition
```
end_.lift (fun x ↦ enrichedHomπ V (Under.forget j ⋙ F₁) (Under.forget j ⋙ F₁) ((Under.map f).obj x)) ⋯ ≫
      end_.π (diagram V (Under.forget j' ⋙ F₁) (Under.forget j' ⋙ F₁)) k
```
is badly typed, with the target of the first morphism being 
```
end_ (diagram V (Under.map f ⋙ Under.forget j ⋙ F₁) (Under.map f ⋙ Under.forget j ⋙ F₁))
```
and the source of the second morphism being
```
end_ (diagram V (Under.forget j' ⋙ F₁) (Under.forget j' ⋙ F₁))
```

Reply:
So maybe there is some defeq in `functorEnrichedHom` that could be lifted to an explicit `eqToHom`? I better move on to other things now, sorry.

Perhaps there is a missing piece of tooling here, however. What about a `#check_comp` tactic that looks for all morphism compositions and reports when sources and targets don't match (at reducible+instances transparency). (Or even more generally, a `#check_implicits`.) I'm not saying that any report from this would be a problem, but it might be helpful in identifying what is going on in problems like this.

Reply:
#21357

Reply:
I have managed to remove all `erw` by introducing `precompEnrichedHom'` which is a variant of `enrichedHom V F₁ F₂ ⟶ enrichedHom V (G ⋙ F₁) (G ⋙ F₂)` but with `G ⋙ F₁` and `G ⋙ F₂` replaced by isomorphic functors. I expected it would complicate things, but eventually automation works better...